src/main/java/org/zalando/nakadi/validation/SchemaEvolutionService.java

```diff
@@ -97,6 +102,12 @@ public EventType evolve(final EventType original, final EventTypeBase eventType)
 
         final Version.Level changeLevel = semanticOfChange(changes, original.getCompatibilityMode());
 
+        if (changeLevel == NO_CHANGES && !original.getSchema().getSchema().equals(eventType.getSchema().getSchema())) {
+            LOG.error("undetected schema changes from {} to {}", original.getSchema().getSchema(),
+                    eventType.getSchema().getSchema());
+            throw new MyNakadiRuntimeException1("undetected schema changes");
```

Comment:
I don't really like it when we throw `MyNakadiRuntimeException1`, because the name doesn't say anything about what problem there was. Also, if we do it a lot, it might be difficult for the caller to find out where it comes from. I prefer to subclass the exception with something that has a clear name